iceoryx_hoofs/source/posix_wrapper/posix_access_rights.cpp

```diff
@@ -147,10 +147,7 @@ PosixUser::groupVector_t PosixUser::getGroups() const noexcept
     }
 
     gid_t userDefaultGroup = getpwnamCall->value->pw_gid;
-
-    /// NOLINTJUSTIFICATION @todo iox-#1614 use upcoming cxx::array
-    /// NOLINTNEXTLINE(hicpp-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    gid_t groups[MaxNumberOfGroups];
+    containers::UninitializedArray<gid_t, MaxNumberOfGroups, containers::OneBuffer> groups;
```

Comment:
@FerdinandSpitzschnueffler

OK this will be a bit lengthy. :-)

This is confusing to a potential reader IMO (even without knowing about the change here).

1) If `gid_t` is an integer type according to the standard but not necessarily unsigned (https://stackoverflow.com/questions/1922761/size-of-pid-t-uid-t-gid-t-on-linux, I did not dig deeper).
Now for initializing with zero this does not matter but with an all one bit pattern it does. 

Assume we can rely on it being an unsigned integer (of 4 bytes, presumably), than we now initialize with the max value but this is not very transparent from the code.

2) It was not done initialized with max before, why is it now? I guess it is to prevent initializing with the root id 0.

According to https://en.wikipedia.org/wiki/Group_identifier, -1 is the special purpose value to indicate no value, or, in twos complement representation, all ones (number of bytes is presumably 4 but not necessarily specified).

So by the using the all one bit policy each group id is initialized to invalid (seems reasonable), but it is pretty much not transparent to a developer reading this. That is why it should be set explicitly to -1 in some loop or with some special purpose ctor that initializes every element.

What would be transparent is something like
`containers::UninitializedArray<gid_t, MaxNumberOfGroups> groups{INVALID_GID};`

But this is a lot of inconvenience for just this special case.

**Suggestion**

Since this all is only done for a POSIX call later, which basically operates on the raw C array, we should just use a value initialized C array here like so:

```
gid_t groups[MaxNumberOfGroups];
std::fill_n(groups, MaxNumberOfGroups, INVALID_GID);
```

or with `-1` directly, if with a little less tranparency. If we do not want to use `std::fill_n`, we would need to use memset (with all one bits/bytes, losing the transparency again) or a loop.

We would need the definition
`constexpr gid_t INVALID_GID = reinterpret_cast<gid_t>(-1);`.
This is compatible with the spec https://en.wikipedia.org/wiki/Group_identifier (Special values).
Of course we need a suppression comment then, but it is fine as the POSIX API "requires" a C-array so it is perfectly justified.

All in all I would prefer removing the all one policy as it is too specific and error-prone to use and use the above C array initialization here.

Alternatively we keep the All One bit policy (still fairly exotic) but for transparency we do use a C array here like suggested.

But https://man7.org/linux/man-pages/man3/getgrouplist.3.html suggests that this is all moot and it `groups` can be initialized with anything as it is just a return buffer, the `getgrouplist` will set it accordingly.

So the efficient and POSIX compliant solution is just to keep it as is with a C array and suppression or use a `UnitializedArray` that is not initialized at all (neither zeros, nor ones). This would be my preferred solution, maybe with a comment that `getgrouplist` initializes `groups`.